src/ol/renderer/canvas/TileLayer.js

```diff
@@ -841,21 +847,26 @@ class CanvasTileLayerRenderer extends CanvasLayerRenderer {
     }
     context.imageSmoothingEnabled = true;
 
-    /**
-     * Here we unconditionally expire the source cache since the renderer maintains
-     * its own cache.
-     * @param {import("../../Map.js").default} map Map.
-     * @param {import("../../Map.js").FrameState} frameState Frame state.
-     */
-    const postRenderFunction = (map, frameState) => {
-      const tileSourceKey = getUid(tileSource);
-      const wantedTiles = frameState.wantedTiles[tileSourceKey];
-      const tilesCount = wantedTiles ? Object.keys(wantedTiles).length : 0;
-      this.updateCacheSize(tilesCount);
-      this.tileCache_.expireCache();
-    };
-
-    frameState.postRenderFunctions.push(postRenderFunction);
+    if (this.renderComplete) {
+      /**
+       * Here we unconditionally expire the source cache since the renderer maintains
+       * its own cache.
```

Comment:
Minor, but I think this comment is no longer correct (or related to what the code actually does).  [Initially](https://github.com/openlayers/openlayers/pull/15568/files#diff-ea22661e28efe5304b2c3a407112dccd7f89002aa2f784fac4fbae08aa75bffdR884-R893), we were managing the source cache here.